### PR TITLE
intermediate fix: deep decompose of types to index search records of deeply nested objects

### PR DESCRIPTION
**Pull request by RohinBhargava**
## Short description of the changes made

- recursively parses fields, to extract deeply nested objects
- fixes bug based on discriminant value not exposing facet to Algolia for proper filtering
- adds some basic UI changes (WIP)

## What was the motivation & context behind this PR?

- learnings from last deployment

## How has this PR been tested?

- To be tested with fdr-dev deployment

### CODE REVIEW

**Review by abvthecity — APPROVED** [review]

File: servers/fdr/src/services/algolia/AlgoliaSearchRecordGenerator.ts [hunk 1]
```diff
@@ -1619,182 +1597,338 @@ export class AlgoliaSearchRecordGenerator {
     private collectReferencedTypesToContentV2(
```
Comment (RohinBhargava) [inline servers/fdr/src/services/algolia/AlgoliaSearchRecordGenerator.ts:1597]: This is probably worth some scrutiny

File: packages/ui/app/src/search/SearchHits.tsx [hunk 2]
```diff
@@ -347,9 +349,21 @@ export const SearchMobileHits: React.FC<PropsWithChildren> = ({ children }) => {
 
 function filterHits(hits: SearchRecord[]) {
     const hitTypeMap = {
-        endpoints: new Set(["endpoint", "endpoint-v2", "endpoint-v3", "webhook-v3", "websocket-v3"]),
-        pages: new Set(["page", "page-v2", "page-v3"]),
-        fields: new Set(["field"]),
+        endpoints: new Set([
+            // "endpoint",
+            // "endpoint-v2",
+            // "endpoint-v3",
```
Comment (abvthecity) [inline packages/ui/app/src/search/SearchHits.tsx]: This is gonna cause old docs to break

File: packages/ui/app/src/search/SearchDialog.tsx [hunk 3]
```diff
@@ -88,6 +88,10 @@ export const SearchSidebar: React.FC<PropsWithChildren<SearchSidebar.Props>> = (
 
         return (
             <InstantSearch searchClient={searchClient} indexName={indexName}>
+                <Configure
+                    // filters="type: 'endpoint-v4'"
```
Comment (RohinBhargava) [inline packages/ui/app/src/search/SearchDialog.tsx:92]: Commented out

File: packages/ui/app/src/search/content/EndpointRecordV4.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
+import type { Algolia } from "@fern-api/fdr-sdk/client/types";
+import { visitDiscriminatedUnion } from "@fern-ui/core-utils";
+import cn from "clsx";
+import { LongArrowDownLeft } from "iconoir-react";
+import type { Hit } from "instantsearch.js";
```
Comment (abvthecity) [inline packages/ui/app/src/search/content/EndpointRecordV4.tsx:5]: Why do we need to bring instantsearch.js in?

File: packages/ui/app/src/search/algolia/AlgoliaSearchDialog.tsx [hunk 5]
```diff
@@ -93,6 +93,10 @@ function FernInstantSearch({ searchClient, indexName, inputRef }: FernInstantSea
     );
     return (
         <InstantSearch searchClient={searchClient} indexName={indexName}>
+            <Configure
+                // filters="type: 'endpoint-v4' OR type: 'websocket-v4' OR type: 'webhook-v4' OR type: 'page-v4' OR type: 'field-v1' OR type: 'markdown-section-v1'"
```
Comment (RohinBhargava) [inline packages/ui/app/src/search/algolia/AlgoliaSearchDialog.tsx:97]: commented out

File: packages/ui/app/src/search/content/EndpointRecordV4.tsx (hunk 4 above)
Comment (RohinBhargava) [reply packages/ui/app/src/search/content/EndpointRecordV4.tsx:5]: For the typing to be happy :/. Should we maybe export a wrapper type from common?

**Pull request merged**